tests/python/relax/backend/clml/test_ops.py

```diff
@@ -0,0 +1,1184 @@
+        (3, 3, (1, 1), (1, 1), (1, 1), 64, (3, 224, 224), False, True, False, False, False),
+        (5, 5, (2, 2), (1, 1), (1, 1), 16, (16, 64, 64), False, True, True, False, False),
+        (7, 7, (3, 3), (2, 2), (1, 1), 32, (3, 224, 224), True, False, True, True, False),
+        (3, 3, (0, 0), (1, 1), (1, 1), 512, (256, 14, 14), True, False, True, False, False),
+        (1, 1, (0, 0), (1, 1), (1, 1), 1024, (512, 7, 7), True, False, True, False, False),
+        (1, 3, (0, 0), (1, 1), (1, 1), 64, (64, 7, 7), True, False, True, False, False),
+        (3, 1, (0, 0), (1, 1), (1, 1), 64, (64, 7, 7), False, True, True, True, False),
+    ],
+)
+def test_conv2d_offload(
+    kernel_h,
+    kernel_w,
+    padding,
+    stride,
+    dilation,
+    out_channels,
+    shape,
+    has_bias,
+    has_bn,
+    has_activation,
+    has_pad,
+    is_depthwise,
+    dtype,
+    rpc,
+):
+    low, high = 0, 1
+    data_shape = (1, *shape)
+    if is_depthwise:
+        groups = data_shape[1] // out_channels
+    else:
+        groups = 1
+    padding = (padding[0], padding[1], padding[0], padding[1])
+
+    weight_format = "IOHW" if is_depthwise else "OIHW"
+    weight_shape = (out_channels, data_shape[1] // groups, kernel_h, kernel_w)
+
+    data = np.random.uniform(low, high, size=data_shape).astype(dtype)
+    weight = np.random.uniform(low, high, size=weight_shape).astype(dtype)
+    bias = np.random.uniform(low, high, size=(1, weight_shape[0], 1, 1)).astype(dtype)
+
+    gamma = np.random.uniform(low, high, size=(weight_shape[0],)).astype(dtype)
+    beta = np.random.uniform(low, high, size=(weight_shape[0],)).astype(dtype)
+    mean = np.random.uniform(low, high, size=(weight_shape[0],)).astype(dtype)
+    variance = np.random.uniform(low, high, size=(weight_shape[0],)).astype(dtype)
+
+    inputs = [data]
+    params_np = {"weight": weight}
+    if has_bias:
+        params_np["bias"] = bias
+    if has_bn:
+        params_np.update({"gamma": gamma, "beta": beta, "mean": mean, "variance": variance})
+
+    mod = get_relax_conv2d_module(
+        data_shape,
+        weight_shape,
+        stride=stride,
+        dilation=dilation,
+        padding=padding,
+        weight_layout=weight_format,
+        groups=groups,
+        dtype=dtype,
+        has_bias=has_bias,
+        has_bn=has_bn,
+        has_activation=has_activation,
+        has_pad=has_pad,
+        is_depthwise=is_depthwise,
+    )
+
+    clml_codegen = get_clml_conv2d_codegen(
+        data_shape,
+        weight_shape,
+        stride=stride,
+        dilation=dilation,
+        padding=padding,
+        weight_layout=weight_format,
+        groups=groups,
+        dtype=dtype,
+        has_bias=has_bias,
+        has_bn=has_bn,
+        has_activation=has_activation,
+        has_pad=has_pad,
+        is_depthwise=is_depthwise,
+    )
+
+    run_compare(mod, inputs, params_np, rpc, clml_codegen)
```

Comment:
i know that run_compare won't run if rpc do not exist, however, it is not obvious from the testcase. Would be great to split into two files

test_clml_codegen.py that calls into verify_codegen only and structural comparison.

test_op_exec_clml_codegen.py that calls into runs the ops. 

This would help us to further categorize tests. 

Thank you!